app/src/main/java/fr/free/nrw/commons/explore/ExploreListRootFragment.java

```diff
@@ -113,13 +114,26 @@ public void onAttach(final Context context) {
 
   @Override
   public void onMediaClicked(int position) {
-    Log.d("deneme8","on media clicked");
+
+    if(!openMediaDetails) {
+      // do not open Media detail if openMediaDetails is false
+      return;
+    }
+    openMediaDetails = false;
     container.setVisibility(View.VISIBLE);
-    ((ExploreFragment)getParentFragment()).tabLayout.setVisibility(View.GONE);
+    ((ExploreFragment) getParentFragment()).tabLayout.setVisibility(View.GONE);
     mediaDetails = new MediaDetailPagerFragment(false, true);
     mediaDetails.showImage(position);
     ((ExploreFragment) getParentFragment()).setScroll(false);
     setFragment(mediaDetails, listFragment);
+
+    // Handler which re-enables the click action on Media
+    new Handler().postDelayed(new Runnable() {
+      @Override
+      public void run() {
+        openMediaDetails = true;
+      }
+    }, 2000);
```

Comment:
I just tried, it fixes the problem.
A drawback is that when opening a picture's media details and pressing back and opening another picture's media details fast enough, the second opening does nothing. That might happen with people who do not know that swapping does the same.

Do you think there might be a better solution than this artificial delay? The scenario here (a list of items that open details) is a common pattern.